### PR TITLE
fix(onboarding): don't open onboarding sidebar if no more tasks

### PR DESCRIPTION
**Pull request by roggenkemper**
this pr fixes an issue where when you create a new project and press the "take me to issues" button, we automatically open the onboarding sidebar even if everything is complete. Now , it will check fi there are any tasks left and only open it if there are no tasks left.

### CODE REVIEW

File: static/app/components/sidebar/index.tsx [hunk 1]
```diff
@@ -32,23 +33,26 @@ import PreferencesStore from 'sentry/stores/preferencesStore';
 import SidebarPanelStore from 'sentry/stores/sidebarPanelStore';
 import {useLegacyStore} from 'sentry/stores/useLegacyStore';
 import {space} from 'sentry/styles/space';
-import {Organization} from 'sentry/types';
+import {Organization, Project} from 'sentry/types';
 import {isDemoWalkthrough} from 'sentry/utils/demoMode';
 import {getDiscoverLandingUrl} from 'sentry/utils/discover/urls';
 import theme from 'sentry/utils/theme';
 import useMedia from 'sentry/utils/useMedia';
+import withProjects from 'sentry/utils/withProjects';
```
Comment (evanpurkhiser) [inline static/app/components/sidebar/index.tsx:41]: Can you `useProjects` instead?

Comment (evanpurkhiser) [reply static/app/components/sidebar/index.tsx:41]: Try and avoid using HoC's on functional components when possible.

File: static/app/components/sidebar/index.tsx [hunk 2]
```diff
@@ -73,13 +76,39 @@ function Sidebar({location, organization}: Props) {
...
+  const {projects: project} = useProjects();
+
   const toggleCollapse = () => {
     const action = collapsed ? showSidebar : hideSidebar;
     action();
   };
 
   const bcl = document.body.classList;
 
+  const useOpenOnboardingSidebar = useMemo(() => {
```
Comment (evanpurkhiser) [inline static/app/components/sidebar/index.tsx:91]: Wait this naming is weird, this isn't a hook now it's just a variable.

My suggestion was to extract this logic all out into a function called `useOpenOnboardingSidebar`

Comment (evanpurkhiser) [reply static/app/components/sidebar/index.tsx:91]: Your hook would encapsulate all this logic, including probably also having the `useEffect` inside of it. This type of thing https://react.dev/learn/reusing-logic-with-custom-hooks

**Review by malwilley — APPROVED** [review]

**Pull request merged**